content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-gitlab-identity.md

```diff
@@ -0,0 +1,423 @@
+  - |
+    # Install chainctl
+    wget -O chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_linux_$(uname -m)"
+    chmod +x chainctl
+    mv chainctl /usr/bin
+
+    # Assume
+    chainctl auth login \
+      --identity-token $ID_TOKEN_1 \
+      --identity <your gitlab identity>
+
+    # Explore
+    chainctl policy ls
+```
+
+Let's go over what this configuration does. 
+
+GitLab requires that pipelines have a shell. To this end, this configuration uses the `wolfi-base` image
+
+Next, this configuration creates a JSON Web Token (JWT) with an [`id_tokens`](https://docs.gitlab.com/ee/ci/yaml/index.html#id_tokens) block that will allow the job to be able to fetch an OIDC token and authenticate with Chainguard Enforce. GitLab requires that any JWTs created in this manner must include an `aud` keyword. In this case, it should align with the `audience` associated with the Chainguard identity created in the `gitlab.tf` file: `https://gitlab.com`.
+
+Following that, the job runs a few commands to download and install `chainctl`. It then uses `chainctl`, the JWT, and the Chainguard identity's `id` value to log in to Chainguard Enforce under the assumed identity. Be sure to replace `<your gitlab identity>` with the identity UIDP you noted down in the previous section.
+
+After logging in, the pipeline is able to run any `chainctl` command under the assumed identity. To test out this ability, this configuration runs the `chainctl policies ls` command to list all the policies associated with the `example-group` group. 
+
+After updating the configuration, commit the changes and the pipeline will run automatically. A status box in the dashboard will let you know whether the pipeline runs successfully.
+
+Click the **View Pipeline** button, and then click the **assume-and-explore** job button to open the job's output from the last run. The following lines should appear near the bottom of this output.
+
+```
+. . .
+                            	ID                         	| 	NAME  	| DESCRIPTION |   MODE    
+------------------------------------------------------------+---------------+-------------+-----------
+  4094ca0d1d52b57c89a8eee9f9c3631db0575b3b/aa02f5371fc4075c | trust-any-cgr |         	| ENFORCED 
+. . .
+```
+
+This indicates that the GitLab CI/CD pipeline did indeed assume the identity and run the `chainctl policy ls` command, returning the policy you created in the `sample.tf` file. 
+
+If you'd like to experiment further with this identity and what the workflow can do with it, there are a few parts of this setup that you can tweak. For instance, if you'd like to give this identity different permissions you can change the role data source to the role you would like to grant.
+
+```
+data "chainguard_roles" "editor" {
+  name = "editor"
+}
+```
+
+You can also edit the pipeline itself to change its behavior. For example, instead of inspecting the policies the identity has access to, you could have the workflow inspect the groups.
```

Comment:
```suggestion
You can also edit the pipeline itself to change its behavior. For example, instead of inspecting the policies the identity has access to, you could have the workflow inspect the groups like this:
```